diff --git a/libs/libvqm/vqm_common.c b/libs/libvqm/vqm_common.c
index 29ff83f6c..888eb37d2 100644
--- a/libs/libvqm/vqm_common.c
+++ b/libs/libvqm/vqm_common.c
@@ -35,7 +35,11 @@
 
 #include <assert.h>
 #include <stdio.h>
-#include <malloc.h>
+#ifndef __GLIBC__
+#    include <stdlib.h>
+#else
+#    include <malloc.h>
+#endif
 #include <string.h>
 #include "vqm_dll.h"
 #include "vqm_common.h"
diff --git a/libs/libvqm/vqm_dll.h b/libs/libvqm/vqm_dll.h
index be56cede1..319e2f7d3 100644
--- a/libs/libvqm/vqm_dll.h
+++ b/libs/libvqm/vqm_dll.h
@@ -101,7 +101,11 @@
 
 
 #include <stdio.h>
-#include <malloc.h>
+#ifndef __GLIBC__
+#    include <stdlib.h>
+#else
+#    include <malloc.h>
+#endif
 
 
 /*****************************************************/
diff --git a/libs/libvqm/vqm_parser.y b/libs/libvqm/vqm_parser.y
index 17dd20feb..588a38b73 100644
--- a/libs/libvqm/vqm_parser.y
+++ b/libs/libvqm/vqm_parser.y
@@ -21,7 +21,9 @@
 #include <stdlib.h>
 #include <stdint.h>
 //#include <crtdbg.h>
-#include <malloc.h>
+#ifdef __GLIBC__
+#    include <malloc.h>
+#endif
 #include <string.h>
 #include <assert.h>
 #include "vqm_dll.h"
